android/src/main/java/com/adyenreactnativesdk/component/instant/IdealFragment.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ */
+
+package com.adyenreactnativesdk.component.instant
+
+import android.util.Log
+import androidx.core.os.bundleOf
+import androidx.fragment.app.FragmentManager
+import com.adyen.checkout.components.core.CheckoutConfiguration
+import com.adyen.checkout.components.core.PaymentMethod
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.ideal.IdealComponent
+import com.adyen.checkout.ideal.IdealComponentState
+import com.adyen.checkout.instant.InstantComponentState
+import com.adyen.checkout.instant.InstantPaymentComponent
+import com.adyen.checkout.sessions.core.CheckoutSession
+import com.adyen.checkout.ui.core.AdyenComponentView
+import com.adyenreactnativesdk.AdyenCheckout
+import com.adyenreactnativesdk.R
+import com.adyenreactnativesdk.component.base.BaseComponentFragment
+import com.adyenreactnativesdk.component.base.ComponentData
+import com.adyenreactnativesdk.component.base.ModuleException
+
+class IdealFragment(
+    private val configuration: CheckoutConfiguration,
```

Comment:
Fragment and Activities should not have constructor params. When the system decides to rebuild the Fragment it will use the constructor without params at all times, so any constructor params you need will not be set.

To workaround this you could do something like this in the `show` method below:
```Kotlin
override fun show(
        fragmentManager: FragmentManager,
        configuration: CheckoutConfiguration,
        paymentMethod: PaymentMethod,
        session: CheckoutSession?
) {
    IdealFragment().apply {
            arguments = bundleOf(key to value, key to value....)
        }
        .show(fragmentManager, TAG)
}
```
and in the fragment you retrieve them with:
```Kotlin
private val something: String = arguments?.get("key")
```

Reply:
This is part of a bigger refactoring. Will address in another PR.